pkg/sql/conn_executor_prepare.go

```diff
@@ -643,3 +644,18 @@ func (ex *connExecutor) isAllowedInAbortedTxn(ast tree.Statement) bool {
 		return false
 	}
 }
+
+// getSQLCursor looks up a SQL Cursor by name and returns a pointer to it,
+// if found.
+func (ex *connExecutor) getSQLCursor(cursorName string) *sqlCursor {
+	cursor := ex.getCursorAccessor().getCursor(cursorName)
+	if cursor == nil {
+		// If an invalid identifier is used as a cursor name, it is stored within the cursor
+		// map as enclosed in double quotes. Attempt to look up the cursor name by enclosing
+		// it in double quotes to cover such a case.
+		// TODO(rimadeodhar): Is there a better way to do this? Do we have any helper methods
+		// for this already?
+		cursor = ex.getCursorAccessor().getCursor(fmt.Sprintf("\"%s\"", cursorName))
```

Comment:
This is incorrect.

You want to use the `tree.Name` type (probably also as argument to `getSQLCusor` and the accessor's `getCursor`, and as datatype for `descCmd.Name` above).  `tree.Name` should take care of quoting for you.

Also don't forget to test identifiers that *contain* a double quote character as well.

Reply:
Also I'm seeing this:
```
//  it is stored within the cursor
// map as enclosed in double quotes.
```

That doesn't seem right. The cursor map should use tree.Name as key type, and the double quotes should not be included. This means that the code that populates the map has done something weird.